### PR TITLE
Ftr convert repo

### PR DESCRIPTION
**Pull request by abhirathb**
Description of changeset: This is adding functions that allow for adding a given git repository residing at a path as a new repository in the database object.

### CODE REVIEW

File: knowledge_repo/repository.py [hunk 1]
```diff
@@ -58,6 +58,34 @@ def for_uris(cls, uri):
...
+    def migrate_to_dbrepo(cls,gitpath,newpath):
+        # A new (supposedly) git repository is going to be uploaded
+        # That must be migrated to the database. 
+        # Inputs : 
+        # gitpath
+        # newpath
+
+        from .repositories.dbrepository import DbKnowledgeRepository
+        newpath = "mysql://abhi:1234@localhost/knowledgerepo:%s"%newpath
+        db_obj = DbKnowledgeRepository(newpath)
```
Comment (GeorgeSabu) [inline knowledge_repo/repository.py]: Database uri should not be in the code add it as an env variable

File: knowledge_repo/app/routes/index.py [hunk 2]
```diff
@@ -54,9 +54,12 @@ def render_index():
 @blueprint.route('/testupload')
 @PageView.logged
 def test_upload():
-    global current_repo
-    repo = current_app.append_repo("3","kr-test")
-    current_repo = repo
+    global current_repo,current_app
+    #repo = current_app.append_repo("3","kr-test")
```
Comment (GeorgeSabu) [inline knowledge_repo/app/routes/index.py]: Remove the commenter code

File: knowledge_repo/app/routes/index.py [hunk 3]
```diff
@@ -54,9 +54,12 @@ def render_index():
...
+    #current_repo = repo
+    dbobj = current_repo.migrate_to_dbrepo("ucla_rnaseq_tcga_analysis","ucla-kr")
```
Comment (GeorgeSabu) [inline knowledge_repo/app/routes/index.py]: There is some kr name hard coded here right that should be removed

File: knowledge_repo/app/app.py [hunk 4]
```diff
@@ -285,8 +286,15 @@ def format_date(date):
     def append_repo(self,name,uri):
```
Comment (sahil21) [inline knowledge_repo/app/app.py]: @abhirathb Where is this function being used? Can't see anywhere in the code where it is.

File: knowledge_repo/app/app.py [hunk 5]
```diff
@@ -285,8 +286,15 @@ def format_date(date):
...
         temp = self.repository
         self.repository = knowledge_repo.KnowledgeRepository.append_for_uri(name,uri,temp)
-        self.db_update_index(check_timeouts=False, force=True, reindex=True)
+        self.db_update_index(check_timeouts=False, force=True )
         return self.repository
+   
+    def append_repo_obj(self,name,obj):
+        temp = self.repository
```
Comment (sahil21) [inline knowledge_repo/app/app.py]: @abhirathb Not a big issue but there is no need for `temp` variable.

File: knowledge_repo/app/routes/polly_api.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    db_path = current_app.config['KR_REPO_DB_PATH'] + ':' + dir_name
+    dbobj  = current_repo.migrate_to_dbrepo(dir_path,db_path)
+    current_repo = current_app.append_repo_obj(dir_name,dbobj)
+    return redirect('/feed')
+    error = 200
+    try:
+        db_path = current_app.config['KR_REPO_DB_PATH'] + ':' + dir_name
+        dbobj  = current_repo.migrate_to_dbrepo(dir_path,db_path)
+        current_repo = current_app.append_repo_obj(dir_name,dbobj)
+    except:
```
Comment (sahil21) [inline knowledge_repo/app/routes/polly_api.py]: @abhirathb Catch specific exceptions.

File: knowledge_repo/app/routes/polly_api.py [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        S3-Path 
+    """
+    global current_repo,current_app
+    path = request.args.get('path')
+    dir_name,dir_path = download_dir(path)
+
+    db_path = current_app.config['KR_REPO_DB_PATH'] + ':' + dir_name
+    dbobj  = current_repo.migrate_to_dbrepo(dir_path,db_path)
+    current_repo = current_app.append_repo_obj(dir_name,dbobj)
+    return redirect('/feed')
```
Comment (sahil21) [inline knowledge_repo/app/routes/polly_api.py]: @abhirathb return in the middle?

File: knowledge_repo/repository.py [hunk 8]
```diff
@@ -58,6 +58,36 @@ def for_uris(cls, uri):
...
+        # Inputs : 
+        # gitpath
+        # newpath
+
+        from .repositories.dbrepository import DbKnowledgeRepository
+        db_obj = DbKnowledgeRepository(newpath)
+        gitkr = cls.for_uri(gitpath)
+        
+        for post in gitkr.posts():
+            post_path  = gitpath + '/' + post.path
```
Comment (sahil21) [inline knowledge_repo/repository.py]: @abhirathb Use os.path.join

**Pull request closed without merging**